fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -153,14 +159,191 @@ public OptExpression optimize(ConnectContext connectContext,
 
         context.getTaskScheduler().executeTasks(rootTaskContext, memo.getRootGroup());
 
-        OptExpression result = extractBestPlan(requiredProperty, memo.getRootGroup());
+        OptExpression result;
+        if (!connectContext.getSessionVariable().isSetUseNthExecPlan()) {
+            result = extractBestPlan(requiredProperty, memo.getRootGroup());
```

Comment:
BTW, if 0 always best plan, why not use `result = extractNthPlan(requiredProperty, memo.getRootGroup(), 0);`?